### PR TITLE
EDM-747: Create Ansible lib

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

File: libs/ui-components/src/components/common/LeaveFormConfirmation.tsx [hunk 1]
```diff
@@ -79,11 +79,16 @@ const BrowserBlocker = ({ lock }: { lock: boolean }) => {
 const LeaveFormConfirmation = () => {
   const { dirty, isSubmitting } = useFormikContext();
   const {
+    appType,
     router: { useBlocker },
   } = useAppContext();
 
   const lock = !isSubmitting && dirty;
 
+  if (appType === 'aap') {
```
Comment (rawagner) [inline libs/ui-components/src/components/common/LeaveFormConfirmation.tsx:88]: even though AAP UI uses new react-router, we cannot (yet) use the `useBlocker` hook as they still use `BrowserRouter` and not data router.

File: libs/ansible/src/hooks/useFetch.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  const patch = React.useCallback(
+    async <R>(kind: string, obj: PatchRequest, abortSignal?: AbortSignal): Promise<R> =>
+      patchData(kind, obj, serviceUrl, applyHeaders, abortSignal),
+    [serviceUrl, applyHeaders],
+  );
+
+  const getWsEndpoint = React.useCallback(
+    () => ({
+      wsEndpoint: '',
```
Comment (rawagner) [inline libs/ansible/src/hooks/useFetch.ts:45]: no ws yet

File: libs/ui-components/package.json [hunk 3]
```diff
@@ -36,10 +47,10 @@
     "@patternfly/react-icons": "^5.0.0",
     "@patternfly/react-styles": "^5.1.2",
     "@patternfly/react-table": "^5.0.0",
-    "i18next": "^21.8.14",
-    "react": "17.0.1",
-    "react-dom": "17.0.1",
-    "react-i18next": "^11.7.3",
+    "i18next": ">=21.8.14",
```
Comment (celdrake) [inline libs/ui-components/package.json]: Should these versions be perhaps more limited?
What versions are we targeting?

File: libs/ansible/src/const.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  [ROUTE.FLEET_EDIT]: '/edge/fleets/edit',
+  [ROUTE.FLEET_IMPORT]: '/edge/fleets/import',
+  [ROUTE.DEVICES]: '/edge/devices',
+  [ROUTE.DEVICE_DETAILS]: '/edge/devices',
+  [ROUTE.DEVICE_EDIT]: '/edge/devices/edit',
+  [ROUTE.REPO_CREATE]: '/edge/repositories/create',
+  [ROUTE.REPO_EDIT]: '/edge/repositories/edit',
+  [ROUTE.REPO_DETAILS]: '/edge/repositories',
+  [ROUTE.REPOSITORIES]: '/edge/repositories',
+  [ROUTE.RESOURCE_SYNC_DETAILS]: '/edge/resourcesyncs',
```
Comment (celdrake) [inline libs/ansible/src/const.ts:17]: I just realised that in `useNavigate` the Routes for Repository details, Fleet details and RS details contain an extra space.

Could you change them in this PR, please?

File: libs/ansible/src/hooks/useFetch.ts (hunk 2 above)
Comment (celdrake) [reply libs/ansible/src/hooks/useFetch.ts:45]: Should we avoid showing the Terminal tab in that case?

**Review by celdrake — COMMENTED** [review]
Great work, just a few small comments

File: libs/ui-components/package.json (hunk 3 above)
Comment (rawagner) [reply libs/ui-components/package.json]: i18next 23.15.2
react 18.3.1
react-dom 18.3.1
react-i18next 15.0.2

I've updated the ranges. Let me know if you want them restricted even more.

File: libs/ansible/src/const.ts (hunk 4 above)
Comment (rawagner) [reply libs/ansible/src/const.ts:17]: Good catch, fixed

File: libs/ansible/src/hooks/useFetch.ts (hunk 2 above)
Comment (rawagner) [reply libs/ansible/src/hooks/useFetch.ts:45]: yep, i've added a prop to hide the tab and it is being used from ansible lib

File: libs/ui-components/package.json (hunk 3 above)
Comment (celdrake) [reply libs/ui-components/package.json]: Looks good now, thanks

**Review by celdrake — APPROVED** [review]

**Pull request merged**